modules/friendlytalkback.js

```diff
@@ -399,15 +403,24 @@ Twinkle.talkback.preview = function(form) {
 	if (tbtarget === 'notice') {
 		page = form.noticeboard.value;
 	} else {
-		// usertalk, other, see
-		page = form.page ? form.page.value : mw.config.get('wgUserName');
+		page = form.page ? form.page.value.trim() : mw.config.get('wgFormattedNamespaces')[mw.config.get('wgNamespaceIds').user] + ':' + mw.config.get('wgUserName');
+
+		// Normalize likely mis-entered items
+		// If the page has a namespace prefix (e.g. User:),
+		// passing a namespace to mw.Title won't change it
+		var p = tbtarget === 'usertalk' ? mw.Title.newFromText(page, 3) : mw.Title.newFromText(page);
+		if (p && p.namespace % 2 === 0 && p.namespace !== 4) {
+			var t = new mw.Title(p.title, p.namespace + 1);
+			page = t.getPrefixedText();
+		}
+
 		if (form.message) {
 			message = form.message.value.trim();
 		}
 	}
 
 	var noticetext = Twinkle.talkback.getNoticeWikitext(tbtarget, page, section, message);
-	form.previewer.beginRender(noticetext, 'User_talk:' + mw.config.get('wgRelevantUserName')); // Force wikitext/correct username
+	form.previewer.beginRender(noticetext, mw.config.get('wgFormattedNamespaces')[mw.config.get('wgNamespaceIds').user_talk] + ':' + mw.config.get('wgRelevantUserName')); // Force wikitext/correct username
```

Comment:
Unnecessary obfuscation. "User_talk:" is a recognised namespace on every MediaWiki installation.